R/lockfile.R

```diff
@@ -6,6 +6,7 @@ renv_lockfile_init <- function(project) {
   lockfile$R        <- renv_lockfile_init_r(project)
   lockfile$Python   <- renv_lockfile_init_python(project)
   lockfile$Packages <- list()
+  lockfile$External <- renv_lockfile_external()
```

Comment:
I think you need to update https://github.com/rstudio/renv/blob/334511e6ee93cc473b943701a155656fe12590e6/R/lockfile.R#L146-L164 as well so that `External` comes last.